src/game/logic/object/locomotor.cpp

```diff
@@ -705,31 +705,442 @@ void Locomotor::Loco_Update_Move_Towards_Position(
+        }
+    }
+}
+
+PhysicsTurningType Locomotor::Rotate_Towards_Position(Object *obj, const Coord3D &position, float *angle)
+{
+    return Rotate_Obj_Around_Loco_Pivot(obj, position, Get_Max_Turn_Rate(obj->Get_Body_Module()->Get_Damage_State()), angle);
+}
+
 void Locomotor::Move_Towards_Position_Legs(
     Object *obj, PhysicsBehavior *physics, const Coord3D &goal_pos, float on_path_dist_to_goal, float desired_speed)
 {
-#ifdef GAME_DLL
-    Call_Method<void, Locomotor, Object *, PhysicsBehavior *, const Coord3D &, float, float>(
-        PICK_ADDRESS(0x004BA750, 0x00750C4E), this, obj, physics, goal_pos, on_path_dist_to_goal, desired_speed);
-#endif
+    if (!Get_Downhill_Only() || obj->Get_Position()->z >= goal_pos.z) {
+        float acceleration = Get_Max_Acceleration(obj->Get_Body_Module()->Get_Damage_State());
+        float maxspeed = Get_Max_Speed_For_Condition(obj->Get_Body_Module()->Get_Damage_State());
+
+        if (desired_speed > maxspeed) {
+            desired_speed = maxspeed;
+        }
+
+        float speed = physics->Get_Forward_Speed_2D();
+        float orientation = obj->Get_Orientation();
+        float angle = GameMath::Atan2(goal_pos.y - obj->Get_Position()->y, goal_pos.x - obj->Get_Position()->x);
+
+        if (m_template->m_wanderWidthFactor != 0.0f) {
+            float wanderangle = DEG_TO_RADF(22.5f) * m_template->m_wanderWidthFactor;
+
+            if (Get_Flag(WANDER_DIRECTION)) {
+                m_wanderAngle = m_wanderAngle + speed * m_wanderLength;
+
+                if (m_wanderAngle > wanderangle) {
+                    Set_Flag(WANDER_DIRECTION, false);
+                }
+            } else {
+                m_wanderAngle = m_wanderAngle - speed * m_wanderLength;
+
+                if (-wanderangle > m_wanderAngle) {
+                    Set_Flag(WANDER_DIRECTION, true);
+                }
+            }
+
+            angle = Normalize_Angle(angle + m_wanderAngle);
+        }
+
+        float angle2 = Normalize_Angle(angle, orientation);
+        Loco_Update_Move_Towards_Angle(obj, angle);
+
+        float angle3 = GameMath::Fabs(angle2) / DEG_TO_RADF(45.f);
+
+        if (angle3 > 1.0f) {
+            angle3 = 1.0f;
+        }
+
+        float minspeed = (1.0f - angle3) * desired_speed;
+        float slowdown = Calc_Slow_Down_Dist(speed, m_template->m_minSpeed, Get_Braking());
+
+        if (on_path_dist_to_goal < slowdown && !Get_Flag(NO_SLOW_DOWN_AS_APPROACHING_DEST)) {
+            minspeed = m_template->m_minSpeed;
+        }
+
+        float speed2 = minspeed - speed;
+
+        if (speed2 != 0.0f) {
+            float mass = physics->Get_Mass();
+            float speed3;
+
+            if (speed2 > 0.0f) {
+                speed3 = acceleration;
+            } else {
+                speed3 = -Get_Braking();
+            }
+
+            float speed4 = mass * speed3;
+            float speed5 = mass * speed2;
+
+            if (GameMath::Fabs(speed5) < GameMath::Fabs(speed4)) {
```

Comment:
This sure is not `GameMath::Fabs(speed5) > GameMath::Fabs(speed4)` ?

Reply:
This is definitely correct.